test/e2e/keycloak_realm_test.go

```diff
@@ -0,0 +1,153 @@
+package e2e
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	keycloakv1alpha1 "github.com/keycloak/keycloak-operator/pkg/apis/keycloak/v1alpha1"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	realmName                  = "test-realm"
+	testOperatorIDPDisplayName = "Test Operator IDP"
+)
+
+func NewKeycloakRealmsCRDTestStruct() *CRDTestStruct {
+	return &CRDTestStruct{
+		prepareEnvironmentSteps: []environmentInitializationStep{
+			prepareKeycloaksCR,
+		},
+		testSteps: map[string]deployedOperatorTestStep{
+			"keycloakRealmBasicTest": {
+				prepareTestEnvironmentSteps: []environmentInitializationStep{
+					prepareKeycloakRealmCR,
+				},
+				testFunction: keycloakRealmBasicTest,
+			},
+			"keycloakRealmWithIdentityProviderTest": {
+				testFunction: keycloakRealmWithIdentityProviderTest,
+			},
+		},
+	}
+}
+
+func getKeycloakRealmCR(namespace string) *keycloakv1alpha1.KeycloakRealm {
+	return &keycloakv1alpha1.KeycloakRealm{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testKeycloakRealmCRName,
+			Namespace: namespace,
+			Labels:    CreateLabel(namespace),
+		},
+		Spec: keycloakv1alpha1.KeycloakRealmSpec{
+			InstanceSelector: &metav1.LabelSelector{
+				MatchLabels: CreateLabel(namespace),
+			},
+			Realm: &keycloakv1alpha1.KeycloakAPIRealm{
+				ID:          realmName,
+				Realm:       realmName,
+				Enabled:     true,
+				DisplayName: "Operator Testing Realm",
+			},
+		},
+	}
+}
+
+func WaitForRealmToBeReady(t *testing.T, framework *framework.Framework, namespace string) error {
```

Comment:
This could probably be moved to the `utils.go` (I believe there are other functions like this as well).